### PR TITLE
webdriverio: adding more addCommand UTs

### PR DESCRIPTION
**Pull request by abjerstedt**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)
Adding UTs to test the scenario described in https://github.com/webdriverio/webdriverio/issues/3479
Note: Code is passing

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/webdriverio/tests/addCommand.test.js [hunk 1]
```diff
@@ -182,6 +185,61 @@ describe('addCommand', () => {
...
+                throw error2
+            })
+
+            await expect(() => browser.function1()).toThrow(error1)
+            await expect(() => browser.function2()).toThrow(error2)
+
+            try {
+                await browser.function1()
+            } catch (error) {
+                expect(error).toBe(error1)
```
Comment (christian-bromann) [inline packages/webdriverio/tests/addCommand.test.js]: there is a chance that this assertion is never being run if the try statement actually runs successfully. Therefor I suggest to put this into a new `test` block and use something like `expect.assertions(number);`

File: packages/webdriverio/tests/addCommand.test.js [hunk 2]
```diff
@@ -182,6 +185,61 @@ describe('addCommand', () => {
...
+            }
+
+            try {
+                await browser.function2()
```
Comment (christian-bromann) [inline packages/webdriverio/tests/addCommand.test.js]: same

File: packages/webdriverio/tests/addCommand.test.js [hunk 3]
```diff
@@ -182,6 +185,61 @@ describe('addCommand', () => {
...
+                await elem.function1()
+            } catch (error) {
+                expect(error).toBe(error1)
+            }
+
+            try {
+                await elem.function2()
+            } catch (error) {
+                expect(error).toBe(error2)
+            }
```
Comment (christian-bromann) [inline packages/webdriverio/tests/addCommand.test.js]: same on the both here too

File: packages/webdriverio/tests/addCommand.test.js (hunk 1 above)
Comment (abjerstedt) [reply packages/webdriverio/tests/addCommand.test.js]: done.

**Pull request merged**